packages/messages/src/browser/notifications-manager.ts

```diff
@@ -66,8 +66,12 @@ export class NotificationManager extends MessageClient {
     protected readonly onUpdatedEmitter = new Emitter<NotificationUpdateEvent>();
     readonly onUpdated = this.onUpdatedEmitter.event;
     protected readonly fireUpdatedEvent = throttle(() => {
-        const notifications = deepClone(Array.from(this.notifications.values()));
-        const toasts = deepClone(Array.from(this.toasts.values()));
+        const notifications = deepClone(Array.from(this.notifications.values()).filter((notification: Notification) =>
```

Comment:
`showProgress`/`reportProgress` should handle it. 

if it's called with empty `message` a log entry should be created to give developers (not users!) a hint.

I wonder how `vscode` does it. if you provide no message in in `showProgress` but in `reportProgress`, especially if `reportProgress` is called with empty message in between. 
does it flicker then? which is my concern this the approach ☝️ 

maybe we should just accept `showProgress` with non-empty message.

Reply:
I didn't want it in showProgress because i wanted to handle empty 'regular' notification as well.
Here, is the place i found that handle all kind of notifications. otherwise i will need to look for empty message in many places (`showProgress`, `reportProgress`, `showMessage` and maybe more).
Can you think about other and better place that centralized all of the notifications?

In vscode (and in Theia with this fix) if no message in `showProgress` it doesn't show the notification until you `reportProgress` with a message. 
if `reportProgress` is called with empty message but with `increment`, the message stays as is and the progress bar updated accordingly. no flickering.

Reply:
>  otherwise i will need to look for empty message in many places (showProgress, reportProgress, showMessage and maybe more).

as far as as I see in `showProgress` and in `showMessage`, yes. that's where they are added.

Reply:
So do you agree with this approach?

Reply:
Not sure yet. We need to check with clients which do not require labels at all, e.g. `git` progress with is handled in the progress locations...

Reply:
My fix is for progress notification.
AFAIK, all git progresses are in the "scm" pane and not as notification. i couldn't found any use of progress notification in Theia.